core/src/main/java/nu/yona/server/crypto/CryptoSession.java

```diff
@@ -30,21 +28,16 @@
 
 public class CryptoSession implements AutoCloseable
 {
+	static final byte CURRENT_CRYPTO_VARIANT_NUMBER = 1;
 	private static final String CIPHER_TYPE = "AES/CBC/PKCS5Padding";
-
-	public interface Executable<T>
-	{
-		T execute();
-	}
-
-	public interface VoidPredicate
-	{
-		boolean test();
-	}
+	private static final String SECRET_KEY_ALGORITHM = "PBKDF2WithHmacSHA256";
+	private static final int SECRET_KEY_LENGTH_BITS = 128;
+	/**
+	 * As we are using secure random passwords, a fixed salt suffices.
```

Comment:
The value of the salt is that passwords automatically become complex and to ensure that two identical passwords to encrypt the same data deliver different results. Given that we use long random passwords, the salts do not add value. If we would use varying salts, we would need to store the salt along with the data, thus bloating it (even more).